crypto/aesgcm_aead.go

```diff
@@ -17,31 +16,34 @@ type aeadAESGCM struct {
 	decrypter cipher.AEAD
 }
 
-// NewAEADAESGCM creates a AEAD using AES-GCM with 12 bytes tag size
-//
-// AES-GCM support is a bit hacky, since the go stdlib does not support 12 byte
-// tag size, and couples the cipher and aes packages closely.
-// See https://github.com/lucas-clemente/aes12.
+var _ AEAD = &aeadAESGCM{}
+
+const ivLen = 12
+
+// NewAEADAESGCM creates a AEAD using AES-GCM
 func NewAEADAESGCM(otherKey []byte, myKey []byte, otherIV []byte, myIV []byte) (AEAD, error) {
-	if len(myKey) != 16 || len(otherKey) != 16 || len(myIV) != 4 || len(otherIV) != 4 {
-		return nil, errors.New("AES-GCM: expected 16-byte keys and 4-byte IVs")
+	// the IVs need to be at least 8 bytes long, otherwise we can't compute the nonce
+	if len(otherIV) != ivLen || len(myIV) != ivLen {
+		return nil, errors.New("AES-GCM: expected 12 byte IVs")
 	}
-	encrypterCipher, err := aes12.NewCipher(myKey)
+
```

Comment:
Is there a reason why you're not checking the key size here (anymore)?

Reply:
Yes, `aes.NewCipher` can be called with different key sizes, so it performs the check if the key has one of the accepted sizes. See https://golang.org/pkg/crypto/aes/#NewCipher

Reply:
Okay, and mint will check for us that we don't accept weak keys?

Reply:
TLS 1.3 doesn't have any weak cipher suites. These are the ones supported by mint: https://github.com/bifurcation/mint/blob/master/common.go#L43-L52.

Reply:
I added tests with 16 and 32 byte keys, and another test that we error on invalid key sizes.